src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -657,6 +657,14 @@ ObjectMonitor::TryLockResult ObjectMonitor::TryLock(JavaThread* current) {
   return first_own == own ? TryLockResult::HasOwner : TryLockResult::Interference;
 }
 
+static void post_monitor_deflate_event(EventJavaMonitorDeflate* event,
+                                       const oop obj) {
+  assert(event != nullptr, "invariant");
+  event->set_monitorClass(obj->klass());
```

Comment:
Now that I have seen the "hidden wait" logic in the other PR, should inflation/deflation events not also check `is_excluded`?

Reply:
Good question. I think if we filter waits on JFR "hidden waits", the inflate and deflate should filter them as well. This requires a bit of reshuffling, but I think the thing I have in new version works.